cdc/cdc/capture/capture.go

```diff
@@ -434,6 +434,27 @@ func (c *Capture) AsyncClose() {
 		c.regionCache.Close()
 		c.regionCache = nil
 	}
+
+	c.cancel()
+
+	if c.etcdClient != nil {
+		timeoutCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		if err := c.election.Resign(timeoutCtx); err != nil {
+			log.Warn("failed to resign", zap.Error(err))
+		}
+		cancel()
+
+		timeoutCtx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+		if err := c.etcdClient.DeleteCaptureInfo(timeoutCtx, c.info.ID); err != nil {
```

Comment:
Why is `DeleteCaptureInfo` moved here?

Reply:
It seems because of the `c.etcdClient = nil` at the end of this function, where is before the original point.

Reply:
yesh!